hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/index/bloom/TestKeyRangeLookupTree.java

```diff
@@ -68,7 +68,7 @@ public void testFileGroupLookUpManyEntriesWithSameStartValue() {
     updateExpectedMatchesToTest(toInsert);
     keyRangeLookupTree.insert(toInsert);
     for (int i = 0; i < 10; i++) {
-      endKey += 1 + RANDOM.nextInt(100);
+      endKey += 1 + RANDOM.nextInt(50);
       toInsert = new KeyRangeNode(startKey, Long.toString(endKey), UUID.randomUUID().toString());
       updateExpectedMatchesToTest(toInsert);
```

Comment:
Can the `KeyRangeNode` be fixed to compare with `Long` instead of `String` here ? The fix is fragile because it is related with the key value range. And we can not always keep ensurence that the key is smaller than `1024`.

Reply:
As `KeyRangeNode` stores recordValue, which is always string value, `KeyRangeNode` doesn't need to compare with other type. I think the test purpose here wants to use Long's order to represent string's order to test `KeyRangeNode` function, so it can work if we force the `endKey` not exceed 1000.

Before the fix, the endKey's maxValue could be 101 * 10 + 250 = 1260, which can exceed 1000. As I forcily set `RANDOM` cannot get value exceed than 50 for each iteration, and the max iteration number is 10, so the endKey cannot exceed 51 * 10 + 250(which is 760), smaller than 1000, so in this range, Long's order is same as the string's order.

Reply:
Yeah, the fix works, it is better if we can fix the record key comparing with Long instead.